### ISSUE TITLE
FN unused_unit: function headers in where clauses

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#5585]
````rust
trait tr {
    fn func<F>(f: F)
    where
        F: FnOnce() -> (); // false negative
}

struct A {}

fn main() {}

impl tr for A {
    fn func<F>(f: F)
    where
        F: FnOnce() -> (), // false negative
    {
    }
}
````
Clippy does not report any of the `-> ()`s.

`clippy 0.0.212 (9912925 2020-05-10)`
